src/Rings/MPolyQuo.jl

```diff
@@ -1056,15 +1054,36 @@ function vector_space(K::AbstractAlgebra.Field, Q::MPolyQuoRing)
   function im(a::Generic.FreeModuleElem)
     @assert parent(a) == V
     b = R(0)
-    for i=1:length(l)
-      c = a[i]
+    for k=1:length(l)
+      c = a[k]
       if !iszero(c)
-        b += c*l[i]
+        b += c*l[k]
       end
     end
     return Q(b)
   end
-  return V, MapFromFunc(im, V, Q)
+
+  # The inverse function. We use the fact that for a chosen monomial ordering 
+  # the monomials which are not in the leading ideal, form a basis for the 
+  # quotient; see Greuel/Pfister "A singular introduction to Commutative Algebra".
+  function prim(a::MPolyQuoRingElem)
+    @assert parent(a) === Q
+    b = lift(a)::MPolyElem
+    o = default_ordering(R)
```

Comment:
Suggestion: Please set ordering somewhere around line1053 and already do a kbase w.r.t. this ordering. If this is not available yet, compute a leading_ideal w.r.t. the ordering and a kbase thereof (for a monomial ideal, the ordering is obviously irrelevant). Then use the same ordering in 1072.

Reply:
The choice of ordering is hidden somewhere in the internals of code that I didn't program. It would be good if whoever did that could just quickly confirm that what I'm doing here is correct.